packages/Chem/src/tests/projects-tests.ts

```diff
@@ -0,0 +1,363 @@
+      ['smiles', 'PAINS (smiles)'], '', true);
+  });
+
+  test('chemical_space', async () => {
+    //column '~smiles.Morgan' is not saved to project since it is an object type column
+    await runSaveAndOpenProjectTest('tests/sar-small_test.csv', runChemicalSpace,
+      ['smiles', 'Embed_X_1', 'Embed_Y_1', 'Cluster (DBSCAN)'], DG.VIEWER.SCATTER_PLOT);
+    //need delay to avoid unhandled exception when calling closeAll()
+    await delay(100);
+  });
+
+  test('chemical_space_sync', async () => {
+    //column '~smiles.Morgan' is not saved to project since it is an object type column
+    await runSaveAndOpenProjectTest('tests/sar-small_test.csv', runChemicalSpace,
+      ['smiles', 'Embed_X_1', 'Embed_Y_1', 'Cluster (DBSCAN)'], DG.VIEWER.SCATTER_PLOT, true);
+    //need delay to avoid unhandled exception when calling closeAll()
+    await delay(100);
+  });
+
+
+  test('activity_cliffs', async () => {
+    //column '~smiles.Morgan' is not saved to project since it is an object type column
+    await runSaveAndOpenProjectTest('activity_cliffs.csv', runActivityCliffs,
+      ['chembl_tid', 'smiles', 'Activity', '~smiles.Morgan', 'Embed_X_1', 'Embed_Y_1', 'sali__1'],
+      DG.VIEWER.SCATTER_PLOT, false, checkActivityCliffsCustomInit);
+    //need delay to avoid unhandled exception when calling closeAll()
+    await delay(100);
+  });
+
+  test('activity_cliffs_sync', async () => {
+    //column '~smiles.Morgan' is not saved to project since it is an object type column
+    await runSaveAndOpenProjectTest('activity_cliffs.csv', runActivityCliffs,
+      ['chembl_tid', 'smiles', 'Activity', '~smiles.Morgan', 'Embed_X_1', 'Embed_Y_1', 'sali__1'],
+      DG.VIEWER.SCATTER_PLOT, true, checkActivityCliffsCustomInit);
+    //need delay to avoid unhandled exception when calling closeAll()
+    await delay(100);
+  });
+
+  after(async () => {
+    grok.shell.closeAll();
+  });
+});
+
+//colList: list of columns which have been added to dataframe by analysis
+//viewerType: viewer which has been created by analysis
+async function runSaveAndOpenProjectTest(tableName: string, analysisFunc: (tv: DG.TableView) => Promise<void>,
+colList: string[], viewerType: string, dataSync?: boolean,
+additionalChecks?: (tv: DG.TableView) => Promise<void>) {
+  let tv;
+  if (dataSync) {
+    await DG.Func.find({ name: 'OpenFile'})[0].prepare({
+      fullPath: `System:AppData/Chem/${tableName}`
+    }).call(undefined, undefined, { processed: false });
+    tv = grok.shell.tv;
+  } else
+    tv = await createTableView(tableName);
+  await delay(100);
+  await analysisFunc(tv);
+  await delay(10);
+  await saveAndOpenProject(tv, dataSync);
+  await delay(10);
+  await dataFrameContainsColumns(colList);
+  if (viewerType)
+    await checkViewerAdded(viewerType);
+  if (additionalChecks)
+    await additionalChecks(tv);
+}
+
+async function runActivityCliffs(tv: DG.TableView): Promise<void> {
+  await DG.Func.find({ package: 'Chem', name: 'activityCliffs' })[0].prepare({
```

Comment:
Why are we using `.prepare.call` if we can do `apply` directly?

Also, what's the reason for calling this function dynamically vs simply calling a JS function?